skdate/src/date.sk

```diff
@@ -0,0 +1,757 @@
+module SKDate;
+
+@cpp_extern("SKIP_localetimezone")
+native fun localeTimeZone(year: UInt32, month: UInt32, day: UInt32): Int32;
+
+@cpp_extern("SKIP_currenttimemillis")
+native fun natCurrentTimeMillis(): Int;
+
+@cpp_extern("SKIP_JS_currenttimemillis")
+native fun jsCurrentTimeMillis(): Float;
+
+fun currentTimeMillis(): Int {
+  if (SkargoVersion.kWasm32) {
+    (jsCurrentTimeMillis() * 1000.0).toInt()
+  } else {
+    natCurrentTimeMillis()
+  }
+}
+
+class File(value: Calendar) extends SKStore.File
+
+const date_maxDaysInMonth: Array<Int> = Array[
+  31,
+  28,
+  31,
+  30,
+  31,
+  30,
+  31,
+  31,
+  30,
+  31,
+  30,
+  31,
+];
+
+class WeekDate(year: Int, week: Int, day: Day) uses Orderable {}
+
+base class Entity {
+  fun toCalendar(): Calendar;
+}
+
+class Date private (
+  private year: Int,
+  private month: Month,
+  private day: UInt8,
+) extends Entity uses Orderable {
+  //
+  static fun create(year: Int, month: Int, day: Int): Date {
+    date = Date(year, January(), UInt8::truncate(1));
+    !date = date.addMonths(month - 1);
+    date.addDays(day - 1);
+  }
+
+  fun toCalendar(): Calendar {
+    Calendar::create(this)
+  }
+
+  fun toNextWeekDay(day: Day): Date {
+    wd = this.getWeekDate();
+    this.addDays(wd.day.toNext(day))
+  }
+
+  fun getMonthStart(): Date {
+    !this.day = UInt8::truncate(1);
+    this
+  }
+
+  fun getYearStart(): Date {
+    !this.day = UInt8::truncate(1);
+    !this.month = January();
+    this
+  }
```

Comment:
Seems like a potential foot-gun for these "get" functions to mutate their receiver?

Reply:
Not in Skip.